streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -703,6 +703,16 @@ void runOnce() {
             // we record the ratio out of the while loop so that the accumulated latency spans over
             // multiple iterations with reasonably large max.num.records and hence is less vulnerable to outliers
             taskManager.recordTaskProcessRatio(totalProcessLatency, now);
+
+            // Only log this at INFO when we actually processed some records to avoid flooding INFO-level logs when
+            // there are no records on the input topics
+            if (totalProcessed > 0) {
+                log.info("Processed {} records for tasks {}, polling for more records now",
```

Comment:
Since this is at the end of runOnce, I'm wondering if it also makes sense to log whether we committed or punctuated, and whether/how many records we polled at the beginning of the method.

Basically, it seems like, if it's a good idea to log some information once per cycle, then it's probably a good idea to summarize everything you'd want to know.

Reply:
Yeah good call. I promoted one of the debug logs you added earlier to log the number of records polled at INFO. I think this should be reasonable but we can always demote it again if it's still showing up too frequently